src/realm/spec.cpp

```diff
@@ -68,6 +69,17 @@ void Spec::init(MemRef mem) noexcept
     update_has_strong_link_columns();
 }
 
+Spec* Spec::get_subspec_by_ndx(size_t subspec_ndx) noexcept
+{
```

Comment:
Also we should assert that the subspec at this index is actually an entry for a subtable since there are entries for links and backlinks in the vector as well.